AutocompleteClientTests/FW/Logic/Worker/ConstructorIOIntegrationTests.swift

```diff
@@ -875,12 +876,187 @@ class ConstructorIOIntegrationTests: XCTestCase {
             let cioError = response.error as? CIOError
             let responseData = response.data!
             let browseResult = responseData.results[0]
-
+            
             XCTAssertNil(cioError)
             XCTAssertNotNil(browseResult)
             XCTAssertEqual(browseResult.labels["is_sponsored"], true)
             expectation.fulfill()
         })
         self.wait(for: expectation)
     }
+    
+    func testBrowseItems() {
+        let expectation = XCTestExpectation(description: "Request 204")
+        let query = CIOBrowseItemsQuery(ids: itemIds)
+        self.constructor.browseItems(forQuery: query, completionHandler: { response in
+            let cioError = response.error as? CIOError
+            XCTAssertNil(cioError)
+            expectation.fulfill()
+        })
+        self.wait(for: expectation)
+    }
+    
+    
+    func testBrowseItemsShouldReturnGroupsWithParentsAndChildren() {
+        let expectation = XCTestExpectation(description: "Request 204")
+        let query = CIOBrowseItemsQuery(ids: itemIds)
+        self.constructor.browseItems(forQuery: query, completionHandler: { response in
+            let cioError = response.error as? CIOError
+            let responseData = response.data!
+            let groups = responseData.groups
+
+            XCTAssertFalse(groups[0].children[0].parents.isEmpty)
+            XCTAssertFalse(groups[0].children.isEmpty)
+            XCTAssertNil(cioError)
+            expectation.fulfill()
+        })
+        self.wait(for: expectation)
+    }
+    
+        func testBrowseItems_WithFilters() {
+            let expectation = XCTestExpectation(description: "Request 204")
+            let facetFilters = [
+                (key: "Brand", value: "A&W")
+            ]
+            let queryFilters = CIOQueryFilters(groupFilter: "101", facetFilters: facetFilters)
+            let query = CIOBrowseItemsQuery(ids: itemIds, filters: queryFilters)
+            self.constructor.browseItems(forQuery: query, completionHandler: { response in
+                let cioError = response.error as? CIOError
+                XCTAssertNil(cioError)
```

Comment:
Same as above, let's add some additional assertions to make sure we're getting back results that was requested for (and accounting for the filters).